src/driver.jl

```diff
@@ -232,26 +253,40 @@ const __llvm_initialized = Ref(false)
 
             # find deferred compiler
             # TODO: recover this information earlier, from the Julia IR
+            #       We can do this now with gpuc.lookup
             worklist = Dict{CompilerJob, Vector{LLVM.CallInst}}()
-            for use in uses(dyn_marker)
-                # decode the call
-                call = user(use)::LLVM.CallInst
-                id = convert(Int, first(operands(call)))
-
-                global deferred_codegen_jobs
-                dyn_val = deferred_codegen_jobs[id]
-
-                # get a job in the appopriate world
-                dyn_job = if dyn_val isa CompilerJob
-                    # trust that the user knows what they're doing
-                    dyn_val
-                else
-                    ft, tt = dyn_val
-                    dyn_src = methodinstance(ft, tt, tls_world_age())
-                    CompilerJob(dyn_src, job.config)
+            if dyn_marker !== nothing
+                for use in uses(dyn_marker)
+                    # decode the call
+                    call = user(use)::LLVM.CallInst
+                    id = convert(Int, first(operands(call)))
+
+                    global deferred_codegen_jobs
+                    dyn_val = deferred_codegen_jobs[id]
+
+                    # get a job in the appopriate world
+                    dyn_job = if dyn_val isa CompilerJob
+                        # trust that the user knows what they're doing
+                        dyn_val
+                    else
+                        ft, tt = dyn_val
+                        dyn_src = methodinstance(ft, tt, tls_world_age())
+                        CompilerJob(dyn_src, job.config)
+                    end
+
+                    push!(get!(worklist, dyn_job, LLVM.CallInst[]), call)
                 end
+            end
 
-                push!(get!(worklist, dyn_job, LLVM.CallInst[]), call)
+            if dyn_marker_v2 !== nothing
+                for use in uses(dyn_marker_v2)
+                    # decode the call
+                    call = user(use)::LLVM.CallInst
+                    dyn_mi = Base.unsafe_pointer_to_objref(
+                        convert(Ptr{Cvoid}, convert(Int, unwrap_constant(operands(call)[1]))))
```

Comment:
I'm confused, I thought doing this at the Julia IR level would allow us to drive deferred compilation sooner instead of having to pry it from the LLVM IR still?

Reply:
I wanted to get the implementation working first. Yes we can now do this sooner.

Reply:
OK, two step approach is fine for me!